presto-pinot-toolkit/src/main/java/com/facebook/presto/pinot/PinotSplit.java

```diff
@@ -133,14 +138,29 @@ public SplitType getSplitType()
         return segments;
     }
 
+    public Optional<String> getGrpcHost()
+    {
```

Comment:
Can you add a comment explaining this logic here. Should the GRPC host be saved as an immutable field in the split, instead of recomputing here ?

Can you use Optional.map instead of the if condition below ...

Reply:
This method is only called once, so I think it's fine to set the logic here.